backend/tournesol/tests/test_api_subsamples.py

```diff
@@ -0,0 +1,184 @@
+        # This second poll allows to check if the API results are correctly
+        # limited by poll.
+        self.extra_poll = PollWithCriteriasFactory.create()
+
+        self.base_subsamples_url = f"/users/me/subsamples/{self.poll1.name}/"
+
+        self.user1 = UserFactory(username=self._username1)
+        self.user2 = UserFactory(username=self._username2)
+
+        self.poll1_videos1 = self._create_contributor_ratings(
+            self.poll1, self.user1, self.user1_ratings_nb
+        )
+
+        self.poll1_videos2 = self._create_contributor_ratings(
+            self.poll1, self.user2, self.user2_ratings_nb
+        )
+
+        self._create_contributor_ratings(self.extra_poll, self.user1, 20)
+
+    def test_authentication_required(self):
+        """
+        Anonymous users cannot access their ratings' sub-samples.
+        """
+        response = self.client.get(self.base_subsamples_url)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        self.client.force_authenticate(self.user1)
+        response = self.client.get(self.base_subsamples_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_response_integrity_user1(self):
+        """
+        An authenticated user can get a sub-sample of its compared entities,
+        order by the individual score (descending) computed for the poll's
+        main criterion.
+        """
+        self.client.force_authenticate(self.user1)
+        response = self.client.get(self.base_subsamples_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        results = response.data["results"]
+
+        # By default, the API returns at most 20 rated entities from 20
+        # different buckets.
+        self.assertEqual(len(results), 20)
+
+        # The results:
+        #  - are ordered by score descending
+        #  - only contains entities rated by the logged-in user
+        for idx, item in enumerate(results):
+            from_ = idx * 2
+            to = from_ + 2
+            self.assertIn(
+                item["entity"]["uid"], [video.uid for video in self.poll1_videos1[from_:to]]
+            )
+            self.assertIn("individual_rating", item)
+            self.assertIn("collective_rating", item)
+            self.assertEqual(item["subsample_metadata"]["bucket"], idx + 1)
+
+    def test_response_integrity_user2_few_ratings(self):
+        """
+        An authenticated user with less than 20 ratings, will get as many
+        items as his/her number of ratings.
+        """
+        self.client.force_authenticate(self.user2)
+        response = self.client.get(self.base_subsamples_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        results = response.data["results"]
+        self.assertEqual(len(results), 4)
+        self.assertEqual(response.data["count"], 4)
+
+        # The results:
+        #  - are ordered by score descending
```

Comment:
I don't think you assert this, but it would be good to check it.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1765/commits/9d23dcb343083fe663062326d8b801ec61e98659